### PR TITLE
Lighter headers

### PR DESCRIPTION
**Pull request by pirapira**
Mostly removing unused functions.

In `libethereum/BlockQueue.h`, a template variable `T` was used for closures, but I replaced it with some `std::function`.

### CODE REVIEW

File: libdevcore/concurrent_queue.h [hunk 1]
```diff
@@ -1,60 +0,0 @@
...
-
-namespace dev
-{
-
-/// Concurrent queue.
-/// You can push and pop elements to/from the queue. Pop will block until the queue is not empty.
-/// The default backend (_QueueT) is std::queue. It can be changed to any type that has
-/// proper push(), pop(), empty() and front() methods.
-template<typename _T, typename _QueueT = std::queue<_T>>
-class concurrent_queue
```
Comment (chfast) [inline libdevcore/concurrent_queue.h]: This is not used anywhere?

Comment (pirapira) [reply libdevcore/concurrent_queue.h]: Never even included.

Comment (chfast) [reply libdevcore/concurrent_queue.h]: Sad to see it going away :(

Comment (chfast) [reply libdevcore/concurrent_queue.h]: This is needed by SmartVM.

**Pull request merged**